### PR TITLE
executor: fix hang when the analyze worker unexpectedly exits

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/48356

Problem Summary:

### What is changed and how it works?

if analyze worker or result handler unexpectedly exit. it is possible to hang in the place of sending task.

image

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/executor/analyze.go [hunk 1]
```diff
@@ -513,12 +519,17 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
...
 			logutil.BgLogger().Error("analyze worker panicked", zap.Any("recover", r), zap.Stack("stack"))
 			metrics.PanicCounter.WithLabelValues(metrics.LabelAnalyze).Inc()
+			var job *statistics.AnalyzeJob
+			if task != nil {
+				job = task.job
+			}
 			resultsCh <- &statistics.AnalyzeResults{
 				Err: getAnalyzePanicErr(r),
-				Job: task.job,
+				Job: job,
```
Comment (hi-rustin) [inline pkg/executor/analyze.go:528]: Can I ask what happens if this task is nil? Does it make sense to pass a zero job?

**Review by hi-rustin — COMMENTED** [review]
Thanks!

File: pkg/executor/analyze.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/executor/analyze.go:528]: In test，it can panic before getting task.

Comment (hawkingrei) [reply pkg/executor/analyze.go]: I revert it.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**